File: ruby/features/rules/rules.feature.rb
# frozen_string_literal: true

Given('the customer has {int} cents') do |money|
  @money = money
  @chocolates = []
end

When('the customer tries to buy a {int} cent chocolate bar') do |price|
  @chocolates << 'Mars' if @money > price
end

Then('the sale should not happen') do
  expect(@chocolates).to be_empty
end

Then('the sale should happen') do
  expect(@chocolates).not_to be_empty
end

File: devkit/samples/rules/rules.feature.ts
import assert from 'assert'
import { Given, When, Then } from '@cucumber/fake-cucumber'

Given('the customer has {int} cents', function (money) {
  this.money = money
})

Given('there are chocolate bars in stock', function () {
  this.stock = ['Mars']
})

Given('there are no chocolate bars in stock', function () {
  this.stock = []
})

When('the customer tries to buy a {int} cent chocolate bar', function (price) {
  if(this.money > price) {
    this.chocolate = this.stock.pop
  }
})

Then('the sale should not happen', function () {
  assert(this.chocolate).toBe(null)
})

Then('the sale should happen', function () {
  assert(this.chocolate).notToBe(null)
})
